hack/lib/etcd.sh

```diff
@@ -20,6 +20,8 @@ ETCD_VERSION=${ETCD_VERSION:-3.4.3}
 ETCD_HOST=${ETCD_HOST:-127.0.0.1}
 ETCD_PORT=${ETCD_PORT:-2379}
 export KUBE_INTEGRATION_ETCD_URL="http://${ETCD_HOST}:${ETCD_PORT}"
+ETCD_UNSUPPORTED_ARCH="$(go env GOHOSTARCH)"
+export ETCD_UNSUPPORTED_ARCH
```

Comment:
why are we supporting kubernetes on an unsupported etcd arch?

Reply:
`hack/lib/init.sh` pulls in `hack/lib/etcd.sh`. this change effectively requires go on the host for many things that did not previously. (as evidenced by CI suddenly failing when it did not before)

imo the uname approach was reasonable.

Reply:
> why are we supporting kubernetes on an unsupported etcd arch?

simply cause this make it possible for us to do something like performance test on the arch except amd64.

Reply:
etcd is a kubernetes required dependency though, surely we should look at getting it supported in etcd first..?

Reply:
> > why are we supporting kubernetes on an unsupported etcd arch?
> 
> simply cause this make it possible for us to do something like performance test on the arch except amd64.

and I think k8s' run on the arch like arm64 is a good thing for k8s, and is good trend we will move forward to.

Reply:
> `hack/lib/init.sh` pulls in `hack/lib/etcd.sh`. this change effectively requires go on the host for many things that did not previously. (as evidenced by CI suddenly failing when it did not before)
> 
> imo the uname approach was reasonable.

thanks for the suggestion, I will consider to revert the change.

Reply:
this should also probably not be done globally anyhow, we do not need to be running this every time we import `hack/lib/init.sh` and creating more globally exported environment variables.

when we execute etcd in integration tests specifically we can set this env there.
other jobs like KIND will never use any of this to touch etcd as far as I know (kubeadm configures etcd instead)